sdk/src/main/java/com/karhoo/sdk/analytics/Payloader.kt

```diff
@@ -144,19 +176,140 @@ class Payloader internal constructor(builder: Builder) {
             return this
         }
 
-        fun paymentFailed(errorMessage: String,
-                          lastFourDigits: String,
-                          date: Date,
-                          amount: Int,
-                          currency: String): Builder {
+        fun cardAuthorisationFailure(
+            errorMessage: String,
+            quoteId: String?,
+            paymentMethodUsed: String,
+            lastFourDigits: String,
+            date: Date,
+            amount: Int,
+            currency: String
+        ): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+
             payload[PAYMENT_ERROR_MESSAGE] = errorMessage
             payload[PAYMENT_CARD_LAST_FOUR_DIGITS] = lastFourDigits
             payload[PAYMENT_AMOUNT] = amount
             payload[PAYMENT_CURRENCY] = currency
+            payload[PAYMENT_METHOD_USED] = paymentMethodUsed
             payload[PAYMENT_DATE] = date.toString()
             return this
         }
 
+        fun checkoutOpened(quoteId: String?): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+
+            return this
+        }
+
+        fun cardAuthorisationSuccess(quoteId: String?): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+
+            return this
+        }
+
+        fun loyaltyStatusRequested(
```

Comment:
missing loyalty mode. this is the currently selected mode when the request was made. possible values: none, earn, burn.

Reply:
also: 
- besides slug, errorMessage, and loyaltyStatus the rest should be mandatory
- why is balanced passed separately from the status? the status contains the balance

Reply:
Removed balance as it was double-passed
Added loyaltyMode (was in the ui-sdk, omitted in the network sdk / payloader)